### ISSUE TITLE
Error when running the script

### ISSUE THREAD

**Issue by Sourabhpatel1** [rcassani/keyboard-fusion-rgb#2]
sourabh at SP in ~/p/app
↪ lsusb
Bus 006 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 005 Device 003: ID 2516:0101 Cooler Master Co., Ltd. MM711 Gaming Mouse
Bus 005 Device 002: ID 2357:0604 TP-Link TP-Link UB500 Adapter
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 002: ID 1044:7a35 Chu Yuen Enterprise Co., Ltd AORUS K9 Optical Keyboard
Bus 001 Device 003: ID 1b1c:0c0a Corsair Hydro Series H115i Liquid CPU Cooler
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
sourabh at SP in ~/p/app
↪ sudo chmod -R 666 /dev/bus/usb/001/004
[sudo] password for sourabh: 
chmod: cannot access '/dev/bus/usb/001/004': No such file or directory
sourabh at SP in ~/p/app
↪ sudo chmod -R 666 /dev/bus/usb/001/002
sourabh at SP in ~/p/app
↪ sudo chmod -R 666 /dev/bus/usb/001/001
sourabh at SP in ~/p/app
↪ /bin/python /home/sourabh/project/app/kb.py
Could not open HID keyboard
Error at send_feature_report()
Traceback (most recent call last):
  File "/home/sourabh/project/app/kb.py", line 4, in <module>
    keyboard.set_static_mode(color_rgb=[0xff, 0x00, 0x00])
  File "/home/sourabh/.local/lib/python3.10/site-packages/keyboard_fusion_rgb.py", line 222, in set_static_mode
    self.clean_configuration()
  File "/home/sourabh/.local/lib/python3.10/site-packages/keyboard_fusion_rgb.py", line 198, in clean_configuration
    buf_rsp.pop(0) # except the first byte
AttributeError: 'NoneType' object has no attribute 'pop'
sourabh at SP in ~/p/app
↪

**Comment by rcassani** [comment]
Please, provide a detailed description every time you submit an issue. Also, add **only** relevant information, clutter in the issue makes it hard to follow. This will make it useful for other people with similar issue, and will help others to solve it. 

> `↪ /bin/python /home/sourabh/project/app/kb.py
>
>  Could not open HID keyboard`

It seems the problem is the connection with the keyboard.

Did you update the code to reflect that your keyboard has a different `idProduct`?
See: https://github.com/rcassani/keyboard-fusion-rgb/blob/a81a13e5f43ea06d4fa903d14e2645dd6cb9fd9e/keyboard_fusion_rgb.py#L27

The information for your keyboard is shown in this line:
> Bus 001 Device 002: ID 1044:7a35 Chu Yuen Enterprise Co., Ltd AORUS K9 Optical Keyboard
* `Bus`: `001`
* `Device` : `002`
* `idVendor` : `1044`
* `idProduct` : `7a35`

**Comment by Sourabhpatel1** [comment]
Here is the script I am running after setting the product id as 0x7a35 and after running chmod for the keyboard :
from keyboard_fusion_rgb import KeyboardFusionRGB

keyboard = KeyboardFusionRGB(layout = 'eng_us')
keyboard.product_id = 0x7A35
keyboard.set_static_mode(color_rgb=[0xff, 0x00, 0x00])

Now I am getting the following Error :
sourabh at SP in ~/p/app
↪ sudo chmod -R 666 /dev/bus/usb/001/002
[sudo] password for sourabh: 
sourabh at SP in ~/p/app
↪ /bin/python /home/sourabh/project/app/kb.py
Error at get_feature_report()
Traceback (most recent call last):
  File "/home/sourabh/project/app/kb.py", line 5, in <module>
    keyboard.set_static_mode(color_rgb=[0xff, 0x00, 0x00])
  File "/home/sourabh/.local/lib/python3.10/site-packages/keyboard_fusion_rgb.py", line 222, in set_static_mode
    self.clean_configuration()
  File "/home/sourabh/.local/lib/python3.10/site-packages/keyboard_fusion_rgb.py", line 198, in clean_configuration
    buf_rsp.pop(0) # except the first byte
AttributeError: 'NoneType' object has no attribute 'pop'

**Comment by rcassani** [comment]
> ` Error at get_feature_report()`

This message is prompted here [here](https://github.com/rcassani/keyboard-fusion-rgb/blob/a81a13e5f43ea06d4fa903d14e2645dd6cb9fd9e/keyboard_fusion_rgb.py#L137)

This seems to have the origin in the fact that your keyboard does not follow the same protocol (a 264 byte package). Either the communication protocol is different for that different model, or the layout (which includes the number of keys)

Is your keyboard ENG-US layout?

**Comment by Sourabhpatel1** [comment]
Yes, here is my layout

sourabh at SP in ~/p/app
↪ setxkbmap -query | grep layout
layout:     us
sourabh

PS: 
I noticed that the key order is hard coded. My keyboard is a full size keyboard and I think my keyboard will have a different key order. That might be causing the issue.  I cant figure out the method you used to make the `en_us_keys` array, maybe i can change the order and see if it works.

Also How can I check communication protocol for my keyboard.
Thanks

**Comment by rcassani** [comment]
> I noticed that the key order is hard coded. 

Yes, it's hard coded, as this is part of the driver's communication protocol

> My keyboard is a full size keyboard and I think my keyboard will have a different key order.

The keyboards controlled by this driver are also full size. If the key order is different you need to write its driver.

> Also How can I check communication protocol for my keyboard.

In brief, you would need to intercept the data the keyboard receives, understand how it is organized and what each byte means, finally and write a program to emulate that communication protocol. Detailed instructions [here](https://www.reddit.com/r/aorus/comments/vm8bzr/comment/ie4f3ue/?utm_source=share&utm_medium=web2x&context=3)
